.github/workflows/deployment.yml

```
@@ -2,50 +2,117 @@ name: Deployment
+          git clone https://github.com/DIRACGrid/DIRAC.git ../DIRAC
       - uses: actions/setup-python@v2
         with:
           python-version: "3.9"
-      - name: Install dependencies
-        run: pip install build readme_renderer diraccfg
+      - name: Installing dependencies
+        run: |
+          python -m pip install \
+              build \
+              python-dateutil \
+              pytz \
+              readme_renderer \
+              requests \
+              setuptools_scm \
+              six
       - name: Validate README for PyPI
-        run: python -m readme_renderer README.rst -o /tmp/README.html
-      - name: Make PEP-440 style release on GitHub
-        id: PEP-440
-        if: github.event_name == 'push'
         run: |
-          TAG_NAME=${GITHUB_REF##*/}
-          NEW_STYLE=$(python -c "import diraccfg; major, minor, patch, pre = diraccfg.parseVersion('${TAG_NAME}'); print(f'{major}.{minor}.{patch}', f'a{pre}' if pre else '', sep='')")
-          echo "Converted ${TAG_NAME} version to ${NEW_STYLE}"
-          echo ::set-output name=tag_name::"v$NEW_STYLE"
-          echo ::set-output name=target_commitish::"$(git rev-parse HEAD)"
-      - name: Publish ${{ steps.PEP-440.outputs.tag_name }} release to GitHub
-        if: github.event_name == 'push'
-        uses: softprops/action-gh-release@v1
-        with:
-          target_commitish: ${{ steps.PEP-440.outputs.target_commitish }}
-          body_path: release.notes
-          tag_name: ${{ steps.PEP-440.outputs.tag_name }}
-      - name: Get ${{ steps.PEP-440.outputs.tag_name }} tag
-        if: github.event_name == 'push'
-        uses: actions/checkout@v2
-        with:
-          ref: ${{ steps.PEP-440.outputs.tag_name }}
+          python -m readme_renderer README.rst -o /tmp/README.html
+      - name: Prepare release notes
+        run: |
+          set -xeuo pipefail
+          IFS=$'\n\t'
+          PREV_VERSION=$(git describe --tags --abbrev=0 --match '*[0-9].[0-9]*' --exclude 'v[0-9]r*' --exclude 'v[0-9][0-9]r*')
+          REFERENCE_BRANCH=${GITHUB_REF#refs/heads/}
+          echo "Making release notes for ${REFERENCE_BRANCH} since ${PREV_VERSION}"
+          ../DIRAC/docs/diracdoctools/scripts/dirac-docs-get-release-notes.py \
+              --token "${{ secrets.GITHUB_TOKEN }}" \
+              --sinceTag "${PREV_VERSION}" \
+              --branches "${REFERENCE_BRANCH}" \
+              --repo "${{ github.repository }}" \
+              > release.notes.new
+          cat release.notes.new
+      - name: Create tag if required
+        id: check-tag
+        run: |
+          set -xeuo pipefail
+          IFS=$'\n\t'
+          if [[ "${{ github.repository }}" == "DIRACGrid/WebAppDIRAC" ]]; then
+            if [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
+              if [[ "${{ github.event.ref }}" =~ ^refs/heads/(integration|rel-v([8-9]|[1-9][0-9]+)\.[0-9]+)$ ]]; then
```

here you mean integration, rel-v5.0, ...?

```suggestion
              if [[ "${{ github.event.ref }}" =~ ^refs/heads/(integration|rel-v([5-9]|[1-9][0-9]+)\.[0-9]+)$ ]]; then
```